java/code/src/com/suse/manager/utils/SaltUtils.java

```diff
@@ -1953,6 +1955,40 @@ public void updateSystemInfo(JsonElement jsonResult, String minionId) {
         });
     }
 
+
+    /**
+     * Update the minion connection path according to master/proxy hostname
+     * @param minion the minion
+     * @param master master/proxy hostname
+     * @return true if the path has changed
+     */
+    public boolean updateMinionConnectionPath(MinionServer minion, String master) {
+        boolean changed = minion.updateServerPaths(master);
+
+        if (changed) {
+            ServerFactory.save(minion);
+
+            // Regenerate the pillar data
+            MinionPillarManager.INSTANCE.generatePillar(minion);
+
+            // push the changed pillar data to the minion
+            saltApi.refreshPillar(new MinionList(minion.getMinionId()));
+
+            ApplyStatesAction action = ActionManager.scheduleApplyStates(minion.getCreator(),
+                    Collections.singletonList(minion.getId()),
+                    Collections.singletonList(ApplyStatesEventMessage.CHANNELS),
+                    new Date());
+            try {
+                TASKOMATIC_API.scheduleActionExecution(action, false);
+            }
+            catch (TaskomaticApiException e) {
+                LOG.error("Could not schedule channels state application");
```

Comment:
Nitpick, but I think something more descriptive would be better.

```suggestion
                LOG.error("Could not schedule channels refresh after proxy change. Old URLs remains on minion " + minion.getMinionId());
```

Reply:
ok, fixed